### ISSUE TITLE
Rename FixedOffsetFrame::SetPoseInBodyFrame() and add GetFixedPoseInParentFrame()

### ISSUE THREAD

**Issue by calderpg-tri** [RobotLocomotion/drake#15638]
From conversation with @joemasterjohn-TRI, `FixedOffsetFrame::SetPoseInBodyFrame()` actually sets the pose in the _parent_ frame, not the body frame, and should be named`SetPoseInParentFrame()` (and corresponding documentation update).

To effectively use `SetPoseInParentFrame()`, a method `GetFixedPoseInParentFrame()` should be added so that the pose in parent frame can be easily determined, rather than using `GetFixedPoseInBodyFrame()` and needing to know the body->parent frame transform.

Assigned high priority as this functionality is used by online calibration at TRI, and a recent switch from having camera frames relative to body frames to relative to other fixed offset frames turned up this issue.

**Comment by jwnimmer-tri** [comment]
Ping @joemasterjohn?

**Comment by jwnimmer-tri** [comment]
Searching Drake for "priority: high" and landed here.  This still seems like a serious problem, where the function is not doing what it claims to do?  What is the plan here?

\CC @sherm1

**Comment by joemasterjohn** [comment]
Sorry @jwnimmer-tri, missed the previous ping. I'll have a patch in for this shortly.

**Comment by jwnimmer-tri** [comment]
@joemasterjohn @rpoyner-tri re-ping?  I just found this TODO in Anzu again today.

**Comment by hongkai-dai** [comment]
Any update on this issue? I would say that this function name is a landmine. I thought that if I call
```cc
frame.SetPoseInBodyFrame(frame.CalcPoseInBodyFrame(context));
```
nothing should change.

**Comment by sherm1** [comment]
Paul @mitiguy can you take a look at this?

**Comment by mitiguy** [comment]
Starting this now.

**Comment by mitiguy** [comment]
PR #18198 resolves issue #15638.
